sdk/python/kfp/dsl/_volume_op.py

```diff
@@ -68,6 +68,37 @@ def __init__(self,
                         if data_source is not one of (str, PipelineParam,
                             V1TypedLocalObjectReference)
         """
+        # condition when action == "delete"
+        if "action" in kwargs and kwargs["action"] == "delete":
```

Comment:
I think it's fine to have `action` defined in `__init__`? Just use `create` as a default value and user can override it with `delete` value.

Reply:
It's a good point. I'll add it in the new revision.